### PR TITLE
Bump Kubernetes Client to 5.x

### PR DESCRIPTION
**Pull request by geoand**

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -174,7 +174,7 @@
         <sentry.version>1.7.30</sentry.version>
         <subethasmtp.version>3.1.7</subethasmtp.version>
         <hibernate-quarkus-local-cache.version>0.1.0</hibernate-quarkus-local-cache.version>
-        <kubernetes-client.version>4.13.0</kubernetes-client.version>
+        <kubernetes-client.version>5.0.0-alpha-2</kubernetes-client.version>
```
Comment (gastaldi) [inline bom/application/pom.xml:177]:
```suggestion
        <kubernetes-client.version>5.0.0-alpha-3</kubernetes-client.version>
```

**Comment by gsmet** [comment]
Rebased and squashed.

@geoand @maxandersen is it something we want in Beta1?

**Comment by geoand** [comment]
I think we can wait - doesn't need to be Beta1

**Comment by rsvoboda** [comment]
I this going to be in Quarkus 1.11? 

Another concern is `5.0.0-alpha-3` version of the client, when is .Final planned for it?

**Comment by geoand** [comment]
Exactly, I wanted to wait for a non-alpha version as well.

@manusa any idea?

**Comment by manusa** [comment]
> Exactly, I wanted to wait for a non-alpha version as well.
> 
> @manusa any idea?

We're almost done with the features for this milestone. @metacosm is still trying to add some improvements regarding experience with Custom Resources.
We're starting to check if everything is OK after upgrading some of our dependent projects too.

The idea is to release final version before 2021-01-06 so it can be part of 1.11. Is this deadline OK? Or should we aim for a sooner date?

**Comment by geoand** [comment]
> The idea is to release final version before 2021-01-06 so it can be part of 1.11. Is this deadline OK? Or should we aim for a sooner date?

Thanks for the info!

I'll let @gsmet and @rsvoboda answer about the deadline.

**Comment by rsvoboda** [comment]
https://github.com/quarkusio/quarkus/wiki/Release-Planning says "Core CR1 on January 6th".

So your plan to release on January 6th is quite tight and there is no community bake time for this upgrade.
That concerns me, as 1.11 is important from prod perspective.

**Comment by manusa** [comment]
Aim is/was to release before January 6th. However, it would be good to know what would be a desirable specific target date (from Quarkus perspective) to be able to make it into the 1.11 CR. Or if we should target 1.12 instead.

As I said, most of what we wanted to include in this version is ready, an earlier release is very much doable. We want to make sure nothing is broken, our main effort right now is in testing.

**Comment by rsvoboda** [comment]
Guillaume is going to release 1.11.0.Beta1 later this week, it would be good to have what's ready included in the build to get some feedback.  @gsmet wdyt?

**Comment by gsmet** [comment]
Beta1 is not here to put more pressure on everyone but to offer the opportunity for some teams to get some feedback early.

I'm fine with a Kubernetes Client upgrade in CR1. Just to be precise: January 6th is the release day so the PR needs to be merged the day before.

**Comment by maxandersen** [comment]
Let's give Kubernetes client the time it needs to get to cr state. It's already on a tight schedule.

**Comment by manusa** [comment]
If we cut a 5.0.0-rc-1 (CR), would that make it into the Beta.1? If so, when would we need to provide this release candidate version ("later this week")?

**Comment by gsmet** [comment]
@manusa I would need the update to be merged in Quarkus by tomorrow evening.

Again, there's no need to rush if it's not ready.

**Comment by manusa** [comment]
> @manusa I would need the update to be merged in Quarkus by tomorrow evening.
> 
> Again, there's no need to rush if it's not ready.

Great thanks. It surely is tight. I'll ping you folks in the unlikely event that things go very well and we manage to release in the morning.

**Comment by manusa** [comment]
Hi @geoand / @gastaldi (not sure who is maintaining this PR)

I cut [v5.0.0-beta-1](https://github.com/fabric8io/kubernetes-client/releases/tag/v5.0.0-beta-1) yesterday. 
We'll be updating our dependent projects with this new version to see everything is still working. We're also starting to thoroughly check nothing was broken. The plan is to release once we made sure everything is OK (< 2021-01-05).

I hope we're still on track for 1.11.

**Comment by geoand** [comment]
Thanks @manusa! 

PR updated

File: bom/application/pom.xml [hunk 2]
```diff
@@ -174,7 +174,7 @@
         <sentry.version>1.7.30</sentry.version>
         <subethasmtp.version>3.1.7</subethasmtp.version>
         <hibernate-quarkus-local-cache.version>0.1.0</hibernate-quarkus-local-cache.version>
-        <kubernetes-client.version>4.13.0</kubernetes-client.version>
+        <kubernetes-client.version>5.0.0-beta-1</kubernetes-client.version>
```
Comment (gastaldi) [inline bom/application/pom.xml:177]:
```suggestion
        <kubernetes-client.version>5.0.0</kubernetes-client.version>
```

**Comment by manusa** [comment]
I was coming by to let you know that we just cut 5.0.0, but George was quicker ;)

Don't know how do you want to proceed with the migration guide (https://groups.google.com/g/quarkus-dev/c/G96XwoUdUug/m/If8DnkWdBQAJ). I'm happy to help in any way I can.

File: bom/application/pom.xml (hunk 2 above)
Comment (geoand) [reply bom/application/pom.xml:177]: Feel free to force push to the PR

**Comment by gastaldi** [comment]
@manusa I updated the Quarkus migration guide with a link to the Kubernetes Client migration guide: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-1.11#kubernetes-client

**Comment by manusa** [comment]
:+1: great, thx!

File: bom/application/pom.xml (hunk 2 above)
Comment (gastaldi) [reply bom/application/pom.xml]: Force pushed. I think this PR is ready to be merged once CI passes

Comment (geoand) [reply bom/application/pom.xml]: Great, thanks!

I agree, let's merge when CI passes

**Review by gastaldi — APPROVED** [review]

**Pull request merged**